TSPL.docc/LanguageGuide/Protocols.md

```diff
@@ -749,124 +755,62 @@ a nonfailable initializer or an implicitly unwrapped failable initializer.
 ## Protocols as Types
 
 Protocols don't actually implement any functionality themselves.
-Nonetheless,
-you can use protocols as a fully fledged types in your code.
-Using a protocol as a type is sometimes called an *existential type*,
-which comes from the phrase
-"there exists a type *T* such that *T* conforms to the protocol".
-
-You can use a protocol in many places where other types are allowed, including:
-
-- As a parameter type or return type in a function, method, or initializer
-- As the type of a constant, variable, or property
-- As the type of items in an array, dictionary, or other container
-
-> Note: Because protocols are types,
-> begin their names with a capital letter
-> (such as `FullyNamed` and `RandomNumberGenerator`)
-> to match the names of other types in Swift
-> (such as `Int`, `String`, and `Double`).
-
-Here's an example of a protocol used as a type:
-
-```swift
-class Dice {
-    let sides: Int
-    let generator: RandomNumberGenerator
-    init(sides: Int, generator: RandomNumberGenerator) {
-        self.sides = sides
-        self.generator = generator
-    }
-    func roll() -> Int {
-        return Int(generator.random() * Double(sides)) + 1
-    }
-}
-```
-
-<!--
-  - test: `protocols`
+Nonetheless, you can use a protocol as a type in your code.
 
-  ```swifttest
-  -> class Dice {
-        let sides: Int
-        let generator: RandomNumberGenerator
-        init(sides: Int, generator: RandomNumberGenerator) {
-           self.sides = sides
-           self.generator = generator
-        }
-        func roll() -> Int {
-           return Int(generator.random() * Double(sides)) + 1
-        }
-     }
-  ```
+The most common way to use a protocol as a type
+is to use a protocol as generic constraint.
+Code with generic constraints can work with
+any type that conforms to the protocol,
+and the specific type is chosen by the code that uses the API.
+For example,
+when you call a function that takes an argument
+and that argument's type is generic,
+the caller chooses the type.
+
+Code with an opaque type
+works with some type that conforms to the protocol.
```

Comment:
Although there were two separate SE proposals that introduced opaque types in the result position and in the parameter position, they're the same kind of type.  It's intentional that TSPL avoids calling them out as "opaque result types" and "opaque parameter types" — we specifically moved away from that wording in 3df767d0595b61d2d359140fef6a4a432aff79ea as part of the original drafting of the Opaque Types chapter (merged into 'main' in 77544059).

If we want to dig into this further, let's start a thread on the [forums](https://forums.swift.org/c/92) to avoid derailing this PR, which is about boxed protocol types, not about opaque types.